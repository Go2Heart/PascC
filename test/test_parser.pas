program test(t1, t2);
const
    n = 1000000;
    p = 1000000007;
var
    var1 : real;
    array1 : array [1..9] of integer;
.