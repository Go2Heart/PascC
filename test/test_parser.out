LexToken(PROGRAM,'program',1,0)
LexToken(ID,'test',1,8)
LexToken(LPAREN,'(',1,12)
LexToken(ID,'t1',1,13)
LexToken(COMMA,',',1,15)
LexToken(ID,'t2',1,17)
LexToken(RPAREN,')',1,19)
LexToken(SEMI,';',1,20)
LexToken(CONST,'const',2,22)
LexToken(ID,'n',3,32)
LexToken(EQU,'=',3,34)
LexToken(ICONST,1000000,3,36)
LexToken(SEMI,';',3,43)
LexToken(ID,'p',4,49)
LexToken(EQU,'=',4,51)
LexToken(ICONST,1000000007,4,53)
LexToken(SEMI,';',4,63)
LexToken(ID,'size',5,69)
LexToken(EQU,'=',5,74)
LexToken(ICONST,10,5,76)
LexToken(SEMI,';',5,78)
LexToken(VAR,'var',6,80)
LexToken(ID,'var1',7,88)
LexToken(COLON,':',7,93)
LexToken(REAL,'real',7,95)
LexToken(SEMI,';',7,99)
LexToken(ID,'array1',8,105)
LexToken(COLON,':',8,112)
LexToken(ARRAY,'array',8,114)
LexToken(LBRACK,'[',8,120)
LexToken(ICONST,1,8,121)
LexToken(DOTDOT,'..',8,122)
LexToken(ICONST,9,8,124)
LexToken(RBRACK,']',8,125)
LexToken(OF,'of',8,127)
LexToken(INTEGER,'integer',8,130)
LexToken(SEMI,';',8,137)
LexToken(BEGIN,'begin',10,140)
LexToken(FOR,'for',11,146)
LexToken(ID,'i',11,150)
LexToken(ASSIGN,':=',11,152)
LexToken(ICONST,1,11,155)
LexToken(TO,'to',11,157)
LexToken(ID,'size',11,160)
LexToken(DO,'do',11,165)
LexToken(ID,'writeln',12,172)
LexToken(LPAREN,'(',12,179)
LexToken(ID,'array1',12,180)
LexToken(LBRACK,'[',12,186)
LexToken(ID,'i',12,187)
LexToken(RBRACK,']',12,188)
LexToken(RPAREN,')',12,189)
LexToken(END,'end',13,191)
LexToken(DOT,'.',13,194)
